### ISSUE TITLE
The visual tracker doesn't work anymore?

### ISSUE THREAD

**Issue by terzag** [BOSSoNe0013/track-o-bot#58]
Some time ago, I disabled the option to display cards tracked in game. Today, I enabled it back but it doesn't seem to work in game anymore. Nothing happens when I hover on the decks.

When the option is enabled, it seems to me that the whole screen is a bit darker when HearthStone starts. I assume it's the overlay being set and showing.

Track-o-Bot compiled as of today.

**Comment by angrylinuxuser** [comment]
Do you have recent version or some older one?

**Comment by terzag** [comment]
No. Just compiled it but didn't kept an older version. And the last time I used the option was more than two monthes ago (when I opened a bug report because I couldn't disable it), so not sure when it stopped working exactly.

**Comment by angrylinuxuser** [comment]
Things have changed now and ~/.Hearthstone directory in not needed any more. Please set up wine prefix and game directory in Settings.

**Comment by terzag** [comment]
Oh, ok, it works after setting the paths.

Is it normal that the screen is slighlty darker when the option is enabled, though?

**Comment by angrylinuxuser** [comment]
No, it might be related to mate's compositor casting shadow under the overlay. Can you set up some kind of a window rule to disable shadows ( for now ) for overlay?

**Comment by angrylinuxuser** [comment]
No, it might be related to mate's compositor casting shadow under the overlay. Can you set up some kind of a window rule to disable shadows ( for now ) for overlay?

**Comment by terzag** [comment]
Looks like you're right: I've just disabled the compositor and the display is fine. I'm not sure it will be easy to fix (I'm using Compton and it's a mess to configure for cases like this) but at least it's not a bug in ToB.

**Comment by angrylinuxuser** [comment]
can you check if tob from tnis branch fixes compton https://github.com/angrylinuxuser/track-o-bot/tree/more_fixes ?

**Comment by terzag** [comment]
Yes, the display seems fine with your version.

**Comment by angrylinuxuser** [comment]
ill send merge request :) later today

**Comment by terzag** [comment]
Nice. Thanks.

**Comment by angrylinuxuser** [comment]
Ok im confused now since i made a blunder. Flag i set im my branch was already set so it should not work. Could you please doble check that my version fixes the issue and the main project does not.

**Comment by terzag** [comment]
Mmm... I'm even more confused as it now displays fine with the main version too. Compton is still running and the tracking option is enabled (and is active in game).

The only thing that changed between when it didn't work and now (with your version or the main one) is that I killed Compton (to check that it was the problem) and restarted it. The only possibility I see is that when it's started at boot, Compton is configured differently than when I launch it manually but I don't see why it would behave like this.

**Comment by terzag** [comment]
Okay, I'm completely lost there but the issue is definitely on my side.

I've closed my session and restarted. Launched HS, display got darker. Quit, killed Compton, restarted it, restarted HS, display is fine. And I don't have the tracking overlay anymore, even after killing Compton again.

**Comment by terzag** [comment]
Well, I think I've pretty much sorted this out. It looks like that when Mate is set to use Compton through the system WM settings, it launches it with a specific set of options and not using the user config file. If set to not use a compositor and I launch Compton manually, it will use my config file. I guess I never noticed the difference because the options Mate sets are similar to the ones I want.

In the first case (Mate options), it's a mess with Track-o-Bot, with the screen getting darker (because the overlay is getting a shadow).
In the second case, it works fine because I removed everything about shadows from my config file and set only minimal options.

So, basically, it's fixed on my side as long as I use my own Compton config but I'm available to do more tests using the Mate config if you want to try to patch the shadowed overlay issue (in case it happens with other compositors than Compton).

**Comment by angrylinuxuser** [comment]
From what i've red compton has issues with shadows so mate guys set up for their default settings some overrides: https://bitbucket.org/ubuntu-mate/mate-tweak/src/22374775981cea31e87332001d10b85ad952bafb/marco-compton?at=master&fileviewer=file-view-default

Put your compton config in ~/.config/marco-compton.conf and compton should start with your settings.

And the issue you had with  disappearing overlay it might be caused by the tob hiding overlay when  hearthstone window loses focus.

Checkout https://wiki.archlinux.org/index.php/Compton. You might find some useful info there.

**Comment by terzag** [comment]
Oh, ok, thanks for the update. I assumed Mate used the regular compton.conf and not a file with a specific name.
